File: demo/src/main/java/com/madon/demo/entity/CompagneFonds.java
package com.madon.demo.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class CompagneFonds {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String ref;
    private String titre;
    private String description;
    private String imagePath;
    private Date dateDebut;
    private Date dateFin;
    private Float montantObjectif;
    private Float montantActuel;
    private int nbBeneficiaire;
    @ManyToOne
    private Association association;
    @JsonIgnore
    @OneToMany(mappedBy = "compagneFonds",fetch = FetchType.LAZY)
    private List<Dons> dons;


}

File: front_end/lib/models/CompagneFonds.dart
class CompagneFonds{
  int id;
  String nom;
  String description;
  String dateDebut;
  String dateFin;
  String image;
  String type;
  String etat;

  CompagneFonds(this.id, this.nom, this.description, this.dateDebut,
      this.dateFin, this.image, this.type, this.etat);
  static CompagneFonds fromJson(Map<String, dynamic> json) {
    return CompagneFonds(
      json['id'],
      json['nom'],
      json['description'],
      json['dateDebut'],
      json['dateFin'],
      json['image'],
      json['type'],
      json['etat'],
    );
  }
}